### ISSUE TITLE
Code inconsistencies (char literals, datatype casing, String, else begin)

### ISSUE THREAD

**Issue by AmigoJack** [andgineer/TRegExpr#280]
```delphi
      OP_EOL2:
        begin
          // \Z matches at the very and + before the final line-break (LF and CR LF)
          if regInput < fInputEnd then
          begin
            if (regInput = fInputEnd - 1) and (regInput^ = #10) then
              begin end
            else
            if (regInput = fInputEnd - 2) and (regInput^ = #13) and ((regInput + 1) ^ = #10) then
              begin end
            else
              Exit;
          end;
        end;
```
```delphi
procedure TRegExpr.SetUseOsLineEndOnReplace(AValue: boolean);
begin
  if fUseOsLineEndOnReplace = AValue then
    Exit;
  fUseOsLineEndOnReplace := AValue;
  if fUseOsLineEndOnReplace then
    fReplaceLineEnd := sLineBreak
  else
    fReplaceLineEnd := #10;
end;
```
In these both parts LF and CR are expressed in decimal notation (`#10` and `#13`). But everywhere else the hexadecimal notation is used (`#$a` and `#$d`). You should also use it here in hexadecimal.

Other inconsistencies:

2. Datatypes `boolean`, `integer`, `byte`, `string` are written in lowercase, but <br>
   datatypes `Cardinal`, `Int64`, `UInt64`, `PWideChar`, `UnicodeString`, `WideString`, `WideChar`, `PAnsiChar`, `AnsiString`, `AnsiChar`, `PLongInt`, `LongInt`, `PtrInt`, `Pointer` are written in correct case. <br>
   Consider also starting the lowercase ones with an upper case (`Boolean`, `Integer`, `Byte`).

3. In `procedure TRegExpr.Error()` you make use of datatype `string`, but it should be your own datatype `RegExprString`.

4. Using `else` in a `case` is by default already a block and doesn't need to have its own `begin` and `end`. Logic wise the `else` is part of the `case`, not its labels. Currently your code looks either like this (`else` is wrongly indented like the labels):
   ```delphi
   case * of
     *:
       ...;
     else
       ...;
   end;
   ```
   or it looks like this (`else` is correctly indented like `case` and `end`):
   ```delphi
   case * of
     *:
       ...;
   else
     ...;
   end;
   ```
   Once you use the superfluous `begin` and `end` in the `else` part:
   ```delphi
    case APtr[i] of
      '-':
        IsOn := False;
        ...
    else
      begin
        Result := False;
        Exit;
      end;
    end;
   ```
   You can remove `begin` and `end`, because it is already treated like that and internally comes out as:
   ```delphi
   begin
     begin
       ...
     end;
   end;
   ```

These are just recommendations to harmonize rivaling techniques used in your code - it doesn't change any functionality.

**Comment by Alexey-T** [comment]
Part if these issues are caused by Embarcadero formatter app. I used it to format the entire unit. 
#10 and #$a - leftovers from old times too IMO.
I don't think we need to fix this

**Comment by Alexey-T** [comment]
I made fixes in my PR #332.

- converted `byte / integer / boolean` to `Byte / Integer / Boolean` (but not `string` ID because it's usually lowercased as keyword)
- indented `case` blocks `else` parts better and removed `begin/end` in `else` parts
